promql/promqltest/testdata/native_histograms.test

```diff
@@ -1373,3 +1373,14 @@ eval instant at 1m histogram_fraction(-Inf, +Inf, histogram_nan)
     expect info msg: PromQL info: input to histogram_fraction has NaN observations, which are excluded from all fractions for metric name "histogram_nan"
     {case="100% NaNs"} 0.0
     {case="20% NaNs"} 0.8
+
+clear
+
+# Carefully chosen interval and range so that the zero point of the count is inside the
+# interpolation period.
+load 1m
+  metric {{schema:0 count:15.0 sum:25.0 buckets:[5 10]}} {{schema:0 count:2490.0 sum:75.0 buckets:[15 2475]}}x55
+
+eval instant at 55m increase(metric[90m])
+  {} {{count:2490 sum:50.303030303030305 counter_reset_hint:gauge buckets:[15 2475]}}
```

Comment:
```suggestion
eval instant at 54m30s increase(metric[54m35s])
  {} {{count:2501.736111111111 sum:50.54012345679012 counter_reset_hint:gauge buckets:[11.52488425925926 2491.628086419753]}}
```

This is an example where the resulting histogram is inconsistent: Sum of buckets is 11.52488425925926 + 2491.628086419753 = 2503.152970679012, but count is 2501.736111111111. I think the `compensateLeft` is problematic here because this is a case where `count` isn't adjusted because it isn't extrapolated below zero.